### PR TITLE
consent bug fixed, peer videos enabled to participant in lobby view

### PR DESCRIPTION
**Pull request by bikemc**
## Description

This PR adds support for displaying the video stream of connected participants in the `Lobby` component. The `Video` component from the `ConnectionTest` module is used to display the video stream.

## Related Issue

#126

### CODE REVIEW

File: frontend/src/pages/ConnectionTest/ConnectionTest.tsx [hunk 1]
```diff
@@ -729,7 +729,7 @@ function ReplaceConnection(props: {
  * @param props.srcObject video and audio source
  * @param props.ignoreAudio if true, audio tracks in `srcObject` will be ignored.
  */
-function Video(props: {
+export function Video(props: {
```
Comment (mertyldrr) [inline frontend/src/pages/ConnectionTest/ConnectionTest.tsx:732]: We have a Video component in the atoms folder. if it is easy to adapt it, would be better to use that one

File: frontend/src/pages/Lobby/Lobby.js [hunk 2]
```diff
@@ -5,72 +5,130 @@ import ListItem from "@mui/material/ListItem";
...
+      }
+      return `UserID: ${peer.summary}`;
+    }
+    return `Peer stream ${index + 1}`;
+  };
   return (
     <>
       <AppToolbar />
-      <ConsentModal />
+      <ConsentModal onConsentGiven={setUserConsent} />
```
Comment (mertyldrr) [inline frontend/src/pages/Lobby/Lobby.js:95]: If we use a separate page for consent, we can remove the modal, but if not, it looks good.

Comment (bikemc) [reply frontend/src/pages/Lobby/Lobby.js:95]: I think that will be updated with Berke's pr so I will just leave this like this

**Comment by bikemc** [comment]
### Video Component
**Added Title Support:** 
Extended the Video component to include an optional title prop. This title prop allows you to specify a title for each video element that will be displayed alongside the video.

**Enhanced Rendering:** 
The Video component now uses a Group container to encapsulate both the video (Image) and the title (Text). This change provides a more organized and intuitive way to display video content with associated titles within the canvas.

**Usage Example:**
```
<Video
  src={videoSource}
  participantData={participantInfo}
  title="Sample Title"
/>
```
### VideoCanvas Component
**Rendering Own Video:**
Added code to render the participant's own video using a Video component. It is uniquely identified with a key and displayed with the title "You."

**Rendering Other Participants' Videos (Participant Usage Enabled):** 
For other connected participants, the code maps through the connectedParticipants array. It checks for the participant's own video based on their id and ensures that it is not duplicated in the rendering. There is a check for peer.summary  to enable VideoCanvas for Participant usage.

**Usage Example:**
```
<VideoCanvas
  connectedParticipants={participants}
  sessionData={sessionInfo}
  ownParticipantId={userParticipantId}
/>
```

### Lobby Component

**Enhanced Component Structure:**
The code has been updated to use a VideoCanvas component, which provides a more structured approach to rendering video elements for both the user and connected participants.

**Comment by VanessaVanessaC** [comment]
Now I can see all participants as the experimenter. However, as a participant I can only see myself. Should it be this way?
Also when two participants join and when I left the experiment as an experimenter and rejoin there are 4 participants now in the experiment (each participant is in the list twice). After 5 seconds the two extra participants disappear and the canvas of the video is black. 
To reproduce the bug: 

1. copy participant link of A and B
2. join as an experimenter
3. join as A and B 
4. wait until they are both visible on the experimenter side
5. leave experiment (pretend you have to go back to copy a link for another participant or so)
6. join experiment again
7. look at the participant list (there should be 4 participants now, A and B twice)
8. wait for a couple of seconds --> video goes black, two participants are left in the list

**Comment by bikemc** [comment]
@VanessaVanessaC 
You should see the other participants from the Lobby page. Have you started the experiment? Because before starting the experiment there should be no connection between participants so they only need to see theirself. After starting the experiment   , they need to see others. It should be like the following after starting the experiment: 
image
image

For the bug I couldn't reproduce it, for me after rejoining there is not a change in participant number, so no double listing. Do you use one of the example experiments or create one yourself?

**Review by mertyldrr — APPROVED** [review]
Worked fine for me, participants could see each other after starting the experiment.

OS: macOS
Browser: Microsoft Edge/Chrome

**Comment by VanessaVanessaC** [comment]
The peer video works now. However, I still get the bug as in my last comment and from the experimenter side the upper part of the participant's video is cut off.

**Review by VanessaVanessaC — APPROVED** [review]

**Pull request merged**